### PR TITLE
IGNITE-15603 Unignore correlated queries script tests

### PR DESCRIPTION
**Pull request by alex-plekhanov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_all_subquery.test [hunk 1]
```diff
@@ -123,6 +132,14 @@ NULL	false
 2	false
 3	false
 
+query IT
+SELECT i, i <> ANY(SELECT i FROM integers WHERE i>10) FROM integers ORDER BY i NULLS FIRST;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_all_subquery.test:136]: Does it actually work? The result seems to be `Actual:   [[1, false], [2, false], [3, false], [null, false]]` . No nulls-first

File: modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_any_subquery.test [hunk 2]
```diff
@@ -79,3 +79,53 @@ NULL	NULL
 1	NULL
 2	NULL
 3	true
+
+query IT
+SELECT i, i <> ANY(SELECT i FROM integers WHERE i>2) FROM integers ORDER BY i NULLS FIRST
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_any_subquery.test:84]: Same here. Actual: Actual:   [[1, true], [2, true], [3, false], [null, null]]. Nulls are last.

File: modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_any_subquery.test [hunk 3]
```diff
@@ -79,3 +79,53 @@ NULL	NULL
...
+SELECT i, i <> ANY(SELECT i FROM integers WHERE i>2 OR i IS NULL) FROM integers ORDER BY i NULLS FIRST
+----
+NULL	NULL
+1	true
+2	true
+3	NULL
+
+# use a bunch of cross products to make bigger data sets (> STANDARD_VECTOR_SIZE)
+query IT
+SELECT i, i = ANY(SELECT i1.i FROM integers i1, integers i2, integers i3, integers i4, integers i5, integers i6 WHERE i1.i IS NOT NULL) FROM integers ORDER BY i NULLS FIRST
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/any_all/test_uncorrelated_any_subquery.test:101]: Same here .As I can see, `NULLS FIRST` doesn't work:

Actual:   [[1, true], [2, false], [3, true], [null, null]]

File: modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test [hunk 4]
```diff
@@ -20,6 +20,24 @@ NULL	NULL
 2	2
 3	3
 
+# implicit join with correlated expression in filter
+query II
+SELECT i, (SELECT s1.i FROM integers s1, integers s2 WHERE s1.i=s2.i AND s1.i=4-i1.i) AS j FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test:25]: OK. Actual fix in #9601

File: modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test [hunk 5]
```diff
@@ -20,6 +20,24 @@ NULL	NULL
...
+----
+NULL	NULL
+1	3
+2	2
+3	1
+
+# join with a correlated expression in the join condition
+query II
+SELECT i, (SELECT s1.i FROM integers s1 INNER JOIN integers s2 ON s1.i=s2.i AND s1.i=4-i1.i) AS j FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test:34]: OK. Actual fix in #9601

File: modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test_ignore [hunk 6]
```diff
@@ -6,7 +6,7 @@
 # Ignore https://issues.apache.org/jira/browse/IGNITE-15593
 # Ignore https://issues.apache.org/jira/browse/IGNITE-15588
 # Ignore https://issues.apache.org/jira/browse/IGNITE-15592
-# Ignore https://issues.apache.org/jira/browse/IGNITE-15603
+# Ignore https://issues.apache.org/jira/browse/IGNITE-16000
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test_ignore:9]: Why not with IGNITE-16000 ?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test [hunk 7]
```diff
@@ -9,17 +9,102 @@ INSERT INTO integers VALUES (1), (2), (3), (NULL)
...
 # this will fail, because "i" is not an aggregate but the SUM(i1.i) turns this query into an aggregate
 statement error
 SELECT i, (SELECT SUM(i1.i)) FROM integers i1;
 
 statement error
 SELECT i+1, (SELECT SUM(i1.i)) FROM integers i1;
+
+# subquery inside aggregation
+query RR
+SELECT SUM(i), SUM((SELECT i FROM integers WHERE i=i1.i)) FROM integers i1;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test:37]: Did it work before the fix?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test [hunk 8]
```diff
@@ -9,17 +9,102 @@ INSERT INTO integers VALUES (1), (2), (3), (NULL)
...
+----
+6.000000	6.000000
+
+# aggregate with correlation in filter
+query II
+SELECT i, (SELECT MIN(i) FROM integers WHERE i>i1.i) FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test:43]: Did it work before the fix?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore [hunk 9]
```diff
@@ -312,25 +312,25 @@ query IT
 SELECT i, (SELECT MIN(i) FROM integers WHERE i=i1.i) >= ALL(SELECT i FROM integers WHERE i IS NOT NULL) FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore:312]: Now it works. Why in the ignored tests?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore [hunk 10]
```diff
@@ -312,25 +312,25 @@ query IT
...
 NULL	NULL
-1	0
-2	0
-3	1
+1	false
+2	false
+3	true
 
 query IT
 SELECT i, (SELECT MIN(i) FROM integers WHERE i<>i1.i) > ANY(SELECT i FROM integers WHERE i IS NOT NULL) FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore:320]: Now it works. Why in the ignored tests?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore [hunk 11]
```diff
@@ -312,25 +312,25 @@ query IT
...
 NULL	NULL
-1	1
-2	0
-3	0
+1	true
+2	false
+3	false
 
 query IT
 SELECT i, NOT((SELECT MIN(i) FROM integers WHERE i<>i1.i) > ANY(SELECT i FROM integers WHERE i IS NOT NULL)) FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test_ignore:328]: Now it works. Why in the ignored tests?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore [hunk 12]
```diff
@@ -1,8 +1,8 @@
 # name: test/sql/subquery/scalar/test_correlated_subquery_cte.test
 # description: Test correlated subqueries
 # group: [scalar]
-# Ignore https://issues.apache.org/jira/browse/IGNITE-15583
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:4]: Not actual?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore [hunk 13]
```diff
@@ -71,19 +66,19 @@ NULL	NULL
 query IT
 SELECT i, EXISTS(WITH i2 AS (SELECT i FROM integers WHERE 1=0 AND i1.i=i) SELECT i FROM i2) AS j FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:67]: nulls first?

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore [hunk 14]
```diff
@@ -71,19 +66,19 @@ NULL	NULL
...
-2	0
-3	0
+NULL	false
+1	false
+2	false
+3	false
 
 # correlated ANY with WHERE clause that is always FALSE
 query IT
 SELECT i, i=ANY(WITH i2 AS (SELECT i FROM integers WHERE 1=0 AND i1.i=i) SELECT i FROM i2) AS j FROM integers i1 ORDER BY i;
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:76]: nulls first?

File: modules/calcite/src/test/sql/subquery/scalar/test_grouped_correlated_subquery.test_ignore [hunk 15]
```diff
@@ -1,8 +1,8 @@
 # name: test/sql/subquery/scalar/test_grouped_correlated_subquery.test
 # description: Test correlated subquery with grouping columns
 # group: [scalar]
-# Ignore https://issues.apache.org/jira/browse/IGNITE-15603
 # Ignore https://issues.apache.org/jira/browse/IGNITE-15607
+# Ignore https://issues.apache.org/jira/browse/IGNITE-16000
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_grouped_correlated_subquery.test_ignore:5]: Do we need to add non-working/ignored tests queries?

File: modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test [hunk 16]
```diff
@@ -101,9 +101,23 @@ SELECT a, (SELECT SUM(b) FROM test tsub WHERE test.a=tsub.a) FROM test
 12	21.000000
 13	22.000000
 
-query II
+statement error
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test:104]: Why statement error? The result seems ok.

File: modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test_ignore [hunk 17]
```diff
@@ -102,29 +102,18 @@ SELECT a, (SELECT SUM(b) FROM test tsub WHERE test.a=tsub.a) FROM test
 12	21.000000
 13	22.000000
 
-query II
+statement error
```
Comment (Vladsz83) [inline modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test_ignore:105]: Why statement error. The request seems not failing

File: modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test_ignore (hunk 6 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_complex_correlated_subquery.test_ignore:9]: The new problem was identified, so I've created a new ticket and added information to this test.

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test (hunk 7 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test:37]: Perhaps. I've unignored all currently working tests in related files.

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test (hunk 8 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_correlated_aggregate_subquery.test:43]: Perhaps. I've unignored all currently working tests in related files.

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore (hunk 12 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:4]: Yes, ticket related to GROUP BY issue, byt there is no GROUP BY tests in this file.

File: modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test (hunk 16 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test:104]: Hm. Don't know why this test was marked as `statement error`. I don't see any problem with this query too. 
Fixed, thanks.

File: modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test_ignore (hunk 17 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_scalar_subquery.test_ignore]: Fixed, thanks

File: modules/calcite/src/test/sql/subquery/scalar/test_grouped_correlated_subquery.test_ignore (hunk 15 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_grouped_correlated_subquery.test_ignore:5]: I don't get this comment. The new problem was identified, the new ticket filled, added information about ticket to this file, but no new tests were added to ignored file.

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore (hunk 13 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:67]: NULLS FIRST here by default

File: modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore (hunk 14 above)
Comment (alex-plekhanov) [reply modules/calcite/src/test/sql/subquery/scalar/test_correlated_subquery_cte.test_ignore:76]: NULLS FIRST here by default

**Comment by Vladsz83** [comment]
Approved

**Review by Vladsz83 — APPROVED** [review]

**Comment by alex-plekhanov** [comment]
Merged to sql-calcite.

**Pull request closed without merging**